### ISSUE TITLE
MongoDB: Sharded database PITR support

### ISSUE THREAD

**Issue by faem** [wal-g/wal-g#1293]
Hello, Is there any plan to support PITR for MongoDB sharded database? If yes, Is there any timeline for that? Thank you!

**Comment by x4m** [comment]
Hi! AFAIK currently you can do PITR for each shard of MongoDB cluster. But if you restore each shard on some timestamp independently you may have some inconsistencies for transactions committed at the specific time.
To resolve this inconsistencies you need extra efforts to scan global time of transactions and decide on which point in this global time you want to restore. We don't have this functionality in WAL-G yet, we want it, but I cannot predict when the community will be able to develop this.

**Comment by faem** [comment]
Yes, independent shard restoration is possible. But configServer restoration is disabled. ref: https://github.com/wal-g/wal-g/blob/master/internal/databases/mongo/oplog/applier.go#L131-L134

**Comment by x4m** [comment]
Maybe @perekalov can clarify why configserver ops are disabled?

**Comment by sayedppqq** [comment]
Hello! any update here?
